TEST: v2.3.x + blas variants

h-vetinari opened the pull request:
Continuing the analysis from #341, #326, #317, #307, #293, #288, #273, #252, #237, #227 & #196. Do not merge.

(5 years of BLAS variant testing 😱...)

h-vetinari committed:
test for all blas variants

h-vetinari committed:
remove non-essential test skips

h-vetinari committed:
increase test suite verbosity

h-vetinari committed:
TST: don't build for osx-arm, no tests are run there

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I failed to even lint the recipe, probably because of a conda-smithy bug :cry:. This likely indicates a problem in your `meta.yaml`, though. To get a traceback to help figure out what's going on, install conda-smithy and run `conda smithy recipe-lint --conda-forge .` from the recipe directory. You can also examine the [workflow logs](https://github.com/conda-forge/conda-forge-webservices/actions/runs/17422904817) for more detail.


<sub>This message was generated by GitHub Actions workflow run [https://github.com/conda-forge/conda-forge-webservices/actions/runs/17422904817](https://github.com/conda-forge/conda-forge-webservices/actions/runs/17422904817). Examine the logs at this URL for more detail.</sub>

conda-forge-admin commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipe/recipe.yaml```) and found it was in an excellent condition.

h-vetinari commented:
@isuruf, any idea why
```
 │ │ Command line: `$BUILD_PREFIX/bin/aarch64-conda-linux-gnu-cc -L$PREFIX/lib $SRC_DIR/builddir/meson-private/tmp_f_tiasm/testfile.c -o $SRC_DIR/builddir/meson-private/tmp_f_tiasm/output.exe -ftree-vectorize -fPIC -fstack-protector-strong -fno-plt -O3 -pipe -isystem $PREFIX/include -fdebug-prefix-map=$SRC_DIR=/usr/local/src/conda/numpy-2.3.5 -fdebug-prefix-map=$PREFIX=/usr/local/src/conda-prefix -DNDEBUG -D_FORTIFY_SOURCE=2 -O2 -isystem $PREFIX/include -D_FILE_OFFSET_BITS=64 -O0 -std=c11 -Wl,-O2 -Wl,--sort-common -Wl,--as-needed -Wl,-z,relro -Wl,-z,now -Wl,--allow-shlib-undefined -Wl,-rpath,$PREFIX/lib -Wl,-rpath-link,$PREFIX/lib` -> 0
 │ │ ../numpy/_core/meson.build:441:46: ERROR: Can not run test applications in this cross environment.
 ```
 might be happening? Obvious candidates are recent changes in the compiler stack, but I wouldn't be able to tell how/why?

rgommers commented:
That's the `long double` check. It indicates that this line is somehow not picked up I think:

https://github.com/conda-forge/numpy-feedstock/blob/69b9a91f9a440e5fa90b72e05aeda9fb9c0cfd8b/recipe/build.sh#L13

h-vetinari committed:
MNT: Re-rendered with conda-smithy 3.54.1 and conda-forge-pinning 2025.12.21.08.15.33

Other tools:
- conda-build 25.11.1
- rattler-build 0.54.0
- rattler-build-conda-compat 1.4.10

h-vetinari commented:
Closing as obsolete
